clang/test/SemaSYCL/reqd-sub-group-size-amd_32.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl-is-device -triple amdgcn-amd-amdhsa -target-cpu gfx1010 -internal-isystem %S/Inputs -std=c++2b -verify %s
+
+// Sub-group size is optimized for 32, warn (and ignore the attribute) if the
```

Comment:
FYI - You're not ignoring the attribute IIUC. It looks like you warn but still attach the attribute. I'm not sure how CodeGen handles this but as far as I can tell, the AST still has the attribute. You need to return early if you want to drop it.